examples/http-proxy/http_proxy.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python3
+"""
+a simple http proxy example
+"""
+import asyncio
+
+import aiohttp
+from aiohttp import web
+from datetime import datetime, timedelta
+import functools
+import pathlib
+import shlex
+import sys
+
+
+from yapapi import (
+    __version__ as yapapi_version,
+)
+from yapapi import Golem
+from yapapi.services import Service, Cluster, ServiceState
+
+from yapapi.payload import vm
+
+examples_dir = pathlib.Path(__file__).resolve().parent.parent
+sys.path.append(str(examples_dir))
+
+from utils import (
+    build_parser,
+    TEXT_COLOR_CYAN,
+    TEXT_COLOR_DEFAULT,
+    TEXT_COLOR_GREEN,
+    TEXT_COLOR_YELLOW,
+    run_golem_example,
+)
+
+STARTING_TIMEOUT = timedelta(minutes=4)
+
+
+# ######## local http server
+
+
+class CurrentInstance:
+    __index = 0
+
+    @classmethod
+    def get_index(cls, count: int):
+        if cls.__index >= count:
+            cls.__index = 0
+        idx = cls.__index
+        cls.__index += 1
+        return idx
+
+
+async def web_root(cluster: Cluster, request: web.Request):
+    print(f"{TEXT_COLOR_GREEN}local HTTP request: {dict(request.query)}{TEXT_COLOR_DEFAULT}")
+
+    idx = CurrentInstance.get_index(len(cluster.instances))
+    instance = cluster.instances[idx]
+    instance_ws = instance.network_node.get_websocket_uri(80)
+    app_key = cluster._engine._api_config.app_key
+
+    print(f"{TEXT_COLOR_GREEN}sending a remote request to {instance}{TEXT_COLOR_DEFAULT}")
+    ws_session = aiohttp.ClientSession()
+    async with ws_session.ws_connect(
+        instance_ws, headers={"Authorization": f"Bearer {app_key}"}
+    ) as ws:
+        await ws.send_str("GET / HTTP/1.0\n\n")
+        headers = await ws.__anext__()
+        print(f"{TEXT_COLOR_GREEN}remote headers: {headers.data} {TEXT_COLOR_DEFAULT}")
+        content = await ws.__anext__()
+        data: bytes = content.data
+        print(f"{TEXT_COLOR_GREEN}remote content: {data} {TEXT_COLOR_DEFAULT}")
+
+        # todo pass headers to the response ? ...
+        response_text = data.decode("utf-8")
+        print(f"{TEXT_COLOR_GREEN}local response: {response_text}{TEXT_COLOR_DEFAULT}")
+
+    await ws_session.close()
```

Comment:
hmm... here, the issue is that the connection params are passed in the connection and not in the session, so there's not much advantage and it would only complicate the code